README.md

```diff
@@ -1516,6 +1517,132 @@ FunctionDeclaration(path) {
 
 ----
 
+# Building Nodes
+
+When writing transformations you'll often want to build up some nodes to insert
+into the AST. As mentioned previously, you can do this using the
+[builder](#builder) methods in the [`babel-types`](#babel-types) package.
+
+The method signature for a builder is simple, the name is the same as the node
+type you want to build, but with the first letter lowercase. For example if you
+wanted to build a `MemberExpression` you would use `t.memberExpression(...)`.
+
+The arguments of these builders is decided by the node definition. There's some
+work that's being done to generate easy-to-read documentation on the
+definitions, but for now they can all be found
+[here](https://github.com/babel/babel/tree/master/packages/babel-types/src/definitions).
+
+A node definition looks like the following:
+
+```js
+defineType("MemberExpression", {
+  builder: ["object", "property", "computed"],
+  visitor: ["object", "property"],
+  aliases: ["Expression", "LVal"],
+  fields: {
+    object: {
+      validate: assertNodeType("Expression")
+    },
+    property: {
+      validate(node, key, val) {
+        let expectedType = node.computed ? "Expression" : "Identifier";
+        assertNodeType(expectedType)(node, key, val);
+      }
+    },
+    computed: {
+      default: false
+    }
+  }
+});
+```
+
+Here you can see all the information about this particular node type, including
+how to build it, traverse it, and validate it.
+
+If you look at the `builder` property, you can see the arguments that will be
+passed to the builder method (`t.memberExpression`).
+
+```js
+builder: ["object", "property", "computed"],
+```
+
+Below that you can see the validation for it. We can see that `object` needs to
+be an `Expression`, `property` either needs to be an `Expression` or an
+`Identifier` depending on if the member expression is `computed` or not and
+`computed` is simply a boolean that defaults to `false`.
```

Comment:
You could potentially explain a few of the methods we use in more detail (you already covered default, nodetype so that's cool).

`assertNodeType`, `assertValueType`, `chain(assertValueType("array"), assertEach(assertNodeType("LVal")))`, passing in a function, `assertOneOf` or just link to https://github.com/babel/babel/blob/master/packages/babel-types/src/definitions/index.js